subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -554,6 +554,151 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
 	return 0;
 }
 
+uint8_t ull_adv_sync_acad_enable(struct lll_adv_sync *lll_sync,
+				 uint8_t acad_len, void **acad)
+{
+	struct pdu_adv_com_ext_adv *ter_com_hdr, *ter_com_hdr_prev;
+	struct pdu_adv_ext_hdr *ter_hdr, ter_hdr_prev;
+	struct pdu_adv *ter_pdu, *ter_pdu_prev;
+	uint8_t *ter_dptr, *ter_dptr_prev;
+	uint16_t ter_len, ter_len_prev;
+	uint8_t acad_len_prev;
+	uint8_t *ad_data;
+	uint8_t ter_idx;
+	uint8_t ad_len;
+
+	/* Get reference to previous tertiary PDU data */
+	ter_pdu_prev = lll_adv_sync_data_peek(lll_sync, NULL);
+	ter_com_hdr_prev = (void *)&ter_pdu_prev->adv_ext_ind;
+	ter_hdr = (void *)ter_com_hdr_prev->ext_hdr_adv_data;
+	ter_hdr_prev = *ter_hdr;
+	ter_dptr_prev = ter_hdr->data;
+
+	/* Get reference to new tertiary PDU data buffer */
+	ter_pdu = lll_adv_sync_data_alloc(lll_sync, NULL, &ter_idx);
+	ter_pdu->type = ter_pdu_prev->type;
+	ter_pdu->rfu = 0U;
+	ter_pdu->chan_sel = 0U;
+	ter_com_hdr = (void *)&ter_pdu->adv_ext_ind;
+	ter_com_hdr->adv_mode = ter_com_hdr_prev->adv_mode;
+	ter_hdr = (void *)ter_com_hdr->ext_hdr_adv_data;
+	ter_dptr = ter_hdr->data;
+	*(uint8_t *)ter_hdr = 0U;
```

Comment:
What is the reasoning behind only setting the first octet to 0, instead of e.g. doing a memset?

Reply:
If extended header is present, the minimum 1 octet is the flags.